src/bucket/BucketList.cpp

```diff
@@ -511,16 +566,92 @@ BucketList::getSize() const
 }
 
 void
-BucketList::addBatch(Application& app, uint32_t currLedger,
-                     uint32_t currLedgerProtocol,
-                     std::vector<LedgerEntry> const& initEntries,
-                     std::vector<LedgerEntry> const& liveEntries,
-                     std::vector<LedgerKey> const& deadEntries)
+HotArchiveBucketList::addBatch(Application& app, uint32_t currLedger,
```

Comment:
Good idea. A note for the reader, I used a parameter pack for the base class `addBatch`. While this isn't really necessary right now since both the `HotArchiveBucketList` and `LiveBucketList` have 3 input vector types, this is not true for the upcoming `ColdArchiveBucketList::addBatch` variant.

Reply:
I feel like this PR is growing into a bit of a C++ feature bingo card but I accept your parameter packs as appropriate nonetheless. obviously I asked for this.